### PR TITLE
[web/task split] split web and task deployment + a few supporting bits

### LINKED ISSUE #1182: [web/task split] AWX operator change to split web and task into separate deployment
## Description
starting point https://github.com/ansible/awx-operator/tree/web-task-split

## TODO:
- [x] split up web and task into separate deployment 
- [x] add rsyslog sidecar to both web and task deployment
- [x] add toleration, web_toleration and task_toleration to allow web and task to be schedule on specific nodes https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/
- [x] add affinity, web_affinity and task_affinity to 
- [x] add node_selector, web_node_selector, task_node_selector
- [x] add replicas, task_replicas, web_replicas in CRD to separately configure web and task if specified
- [x] https://github.com/ansible/awx-operator/blob/web-task-split/roles/installer/tasks/main.yml/#L2-L8 make this respect both web and task deployment
- [x] add topology_spread_constraint and task_toplogy_spread_constraints and web_toplogy_spread_constraints on the web deployment
- [x] figure out if project dir need to be on web container [yes/no]
- [ ] should we allow task container to have a larger resource request/limit? because it will likely use more 
- [x] https://github.com/ansible/awx-operator/blob/devel/config/manifests/bases/awx-operator.clusterserviceversion.yaml update CSV with all the new paramaters
- [x] Document new paramaters and change in behavior in README

OPTIONAL changes
- [ ] create new  serviceaccount/role/rolebinding  for web deployment

### PR DESCRIPTION
**Pull request by thedoubl3j**
##### SUMMARY

Fixes portions of #1182

Specifically:
- splitting web and task containers into 2 deployments
- adding rsyslog sidecar container to each deployment
- add toleration, web_toleration and task_toleration to allow web and task to be schedule on specific nodes
- add affinity, web_affinity and task_affinity 
- add node_selector, web_node_selector, task_node_selector

##### ISSUE TYPE

 - New or Enhanced Feature
##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

File: roles/installer/templates/deployments/task.yaml.j2 [hunk 1]
```diff
@@ -269,6 +265,43 @@ spec:
 {% endif %}
 {% if ee_extra_env -%}
             {{ ee_extra_env | indent(width=12, first=True) }}
+{% endif %}
+        - image: '{{ _image }}'
```
Comment (erz4) [inline roles/installer/templates/deployments/task.yaml.j2:269]: any way to add some probes?

Comment (TheRealHaoLiu) [reply roles/installer/templates/deployments/task.yaml.j2:269]: liveness and readiness probe?

File: roles/installer/defaults/main.yml [hunk 2]
```diff
@@ -89,6 +89,22 @@ topology_spread_constraints: ''
...
+# Add node tolerations for the task pods. Specify as literal block. E.g.:
+# tolerations: |
+#   - key: "dedicated"
+#     operator: "Equal"
+#     value: "AWX"
+#     effect: "NoSchedule"
+task_tolerations: ''
+
+# Add node tolerations for the web pods. Specify as literal block. E.g.:
+# tolerations: |
```
Comment (TheRealHaoLiu) [inline roles/installer/defaults/main.yml]: web_tolerations

File: roles/installer/defaults/main.yml [hunk 3]
```diff
@@ -89,6 +89,22 @@ topology_spread_constraints: ''
...
+#   - key: "dedicated"
+#     operator: "Equal"
+#     value: "AWX"
```
Comment (TheRealHaoLiu) [inline roles/installer/defaults/main.yml]: value: "AWXWeb"

File: roles/installer/templates/deployments/web.yaml.j2 [hunk 4]
```diff
@@ -190,6 +186,47 @@ spec:
             {{ web_extra_env | indent(width=12, first=True) }}
 {% endif %}
           resources: {{ web_resource_requirements }}
+        - image: '{{ _image }}'
+          name: '{{ ansible_operator_meta.name }}-rsyslog'
+{% if rsyslog_command %}
+          command: {{ rsyslog_command }}
```
Comment (Zokormazo) [inline roles/installer/templates/deployments/web.yaml.j2:192]: will we ever reach here without `rsyslog_command` or `rsyslog_args` ? what happens in that case? we launch a container with the default command/arg called `myawx-rsyslog` ? I perhaps lost the context here, but these two conditionals, either should not exist or should condition the full existence of this container in the list and not just `command` or `args` fields imo

File: roles/installer/tasks/resources_configuration.yml [hunk 5]
```diff
@@ -250,6 +250,8 @@
         tower_pod_name: '{{ _new_pod["resources"][0]["metadata"]["name"] }}'
   when:
     - tower_resources_result.changed or this_deployment_result.changed
+- debug:
```
Comment (TheRealHaoLiu) [inline roles/installer/tasks/resources_configuration.yml:253]: should this be here

**Review by TheRealHaoLiu — APPROVED** [review]

File: roles/installer/tasks/resources_configuration.yml (hunk 5 above)
Comment (thedoubl3j) [reply roles/installer/tasks/resources_configuration.yml:253]: no, will remove in follow up PR

**Pull request merged**